binaries/cli/src/command/topic/selector.rs

```diff
@@ -0,0 +1,161 @@
+use std::{borrow::Cow, collections::HashMap, fmt, net::IpAddr};
+
+use crate::{
+    LOCALHOST,
+    common::{connect_to_coordinator, resolve_dataflow_identifier},
+};
+use communication_layer_request_reply::TcpRequestReplyConnection;
+use dora_core::{
+    config::InputMapping, descriptor::Descriptor, topics::DORA_COORDINATOR_PORT_CONTROL_DEFAULT,
+};
+use dora_message::{
+    DataflowId,
+    cli_to_coordinator::ControlRequest,
+    coordinator_to_cli::ControlRequestReply,
+    id::{DataId, NodeId},
+};
+use eyre::{Context, ContextCompat, bail};
+use uuid::Uuid;
+
+#[derive(Debug, clap::Args)]
+pub struct DataflowSelector {
+    /// Identifier of the dataflow
+    #[clap(long, short, value_name = "UUID_OR_NAME")]
+    pub dataflow: Option<String>,
+    /// Address of the dora coordinator
+    #[clap(long, value_name = "IP", default_value_t = LOCALHOST)]
+    pub coordinator_addr: IpAddr,
+    /// Port number of the coordinator control server
+    #[clap(long, value_name = "PORT", default_value_t = DORA_COORDINATOR_PORT_CONTROL_DEFAULT)]
+    pub coordinator_port: u16,
+}
+
+impl DataflowSelector {
+    pub fn resolve(&self) -> eyre::Result<(Box<TcpRequestReplyConnection>, Uuid, Descriptor)> {
```

Comment:
Having the session as one of the return times is a bit cumbersome. I think it would be cleaner to instead take a `&mut TcpRequestReplyConnection` argument. Then we could also call this function multiple times (if required in the future) without reconnecting to the coordinator every time.